src/server/system_services/pool_controllers.js

```diff
@@ -68,88 +67,48 @@ class ManagedStatefulSetPoolController extends PoolController {
+            await KubeStore.instance.patch_backingstore(this.pool_name, { spec: { pvPool: { numVolumes: host_count } } });
         } else {
-            await kube_utils.delete_resource('StatefulSet', _statefulset_name);
-        }
-
-        if (delta < 0) {
-            // If we are scaling down we should also clean the pvc resource claimed by
-            // the terminated pods.
-            const pods = await this._list_pods();
-            await Promise.all(pods
-                .filter(pod => {
-                    const pod_name = pod.metadata.name;
-                    const pod_index = Number(pod_name.slice(pod_name.lastIndexOf('-') + 1));
-                    return pod_index >= host_count;
-                })
-                .map(async pod => {
-                    const pod_name = pod.metadata.name;
-
-                    // Wait for the pod to be deleted.
-                    dbg.log0(`ManagedStatefulSetPoolController::scale: waiting for pod ${pod_name} to be deleted`);
-                    await kube_utils.wait_for_delete('pod', pod_name, 15 * 60 * 1000);
-
-                    // Delete the pvc claimed by the pod.
-                    const pvc_name = `noobaastorage-${pod_name}`;
-                    dbg.log0(`ManagedStatefulSetPoolController::scale: deleting pvc ${pvc_name}`);
-                    return kube_utils.delete_resource('pvc', pvc_name);
-                })
-           );
+            await KubeStore.instance.delete_backingstore(this.pool_name);
         }
     }
 
     async upgrade(image) {
-        dbg.log0(`ManagedStatefulSetPoolController::upgrade: upgrading (if needed) stateful set ${
-            this._statefulset_name
-        } to use ${
-            image
-        } image`);
-
-        await kube_utils.patch_resource(
-            'StatefulSet',
-            this._statefulset_name,
-            { spec: { template: { spec: { containers: [{ name: 'noobaa-agent', image }] } } } }
-        );
-
-        const pods = await this._list_pods();
-        await Promise.all(pods.map(async pod => {
-            const pod_name = pod.metadata.name;
-            return kube_utils.wait_for_condition('pod', pod_name, 'ready', 15 * 60 * 1000);
-        }));
+        // TODO: verify done in the operator
     }
 
     async delete() {
         this.scale(0);
     }
 
-    async _list_pods() {
-        const escaped_statefulset_name = string_utils.escape_reg_exp(this._statefulset_name);
-        const pod_name_regexp = new RegExp(`^${escaped_statefulset_name}-\\d+$`);
-        const { items: pods } = await kube_utils.list_resources('pod', 'noobaa-module=noobaa-agent');
-        return pods.filter(pod => pod_name_regexp.test(pod.metadata.name));
+    async get_current_volume_number() {
+        const backingstore = await KubeStore.instance.read_backingstore(this.pool_name);
+        return backingstore.spec.pvPool.numVolumes;
```

Comment:
Do not read the spec. It does not say anything about the current state. you should read the status of the backing store  where the backing store should save the number of actual volumes (pods) that are up and running

Reply:
the operator is now in control - current state is not interesting - we just need to check if the operator and core holds different numbers and update the operator if so